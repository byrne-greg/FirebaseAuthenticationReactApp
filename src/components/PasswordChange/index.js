import PasswordChangePage, { PasswordChangeForm } from "./PasswordChange";

export default PasswordChangePage;
export { PasswordChangeForm };
