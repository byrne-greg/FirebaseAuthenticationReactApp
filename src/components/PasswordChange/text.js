export default {
  changeYourPassword: "Change your password",
  resetMyPassword: "Reset My Password",
  newPassword: "New Password",
  confirmNewPassword: "Confirm New Password"
};
