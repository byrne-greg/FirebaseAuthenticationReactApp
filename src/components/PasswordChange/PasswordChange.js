import React, { useState, useContext } from "react";
import { FirebaseContext } from "../Firebase";
import text from "./text";

const PasswordChangePage = () => (
  <div>
    <h1>{text.changePassword}</h1>
    <PasswordChangeForm />
  </div>
);

const NewPasswordInput = ({ newPassword, onNewPasswordChange }) => (
  <>
    <label id="new-password-label">{text.newPassword}</label>
    <input
      aria-labelledby="new-password-label"
      name="new-password"
      value={newPassword}
      onChange={onNewPasswordChange}
      type="password"
      placeholder={text.newPassword}
    />
  </>
);

const ConfirmNewPasswordInput = ({
  confirmNewPassword,
  onConfirmNewPasswordChange
}) => (
  <>
    <label id="confirm-new-password-label">{text.confirmNewPassword}</label>
    <input
      aria-labelledby="confirm-new-password-label"
      name="confirm-new-password"
      value={confirmNewPassword}
      onChange={onConfirmNewPasswordChange}
      type="password"
      placeholder={text.confirmNewPassword}
    />
  </>
);

const SubmitButton = ({ isInvalid }) => (
  <button disabled={isInvalid} type="submit">
    {text.changePassword}
  </button>
);

const ErrorAlert = ({ errorText }) => (
  <div id="submit-error">
    <p>{errorText}</p>
  </div>
);

const PasswordChangeForm = () => {
  const [newPassword, setNewPassword] = useState("");
  const [confirmNewPassword, setConfirmNewPassword] = useState("");
  const [error, setError] = useState(null);
  const firebase = useContext(FirebaseContext);

  const isInvalid = !newPassword || newPassword !== confirmNewPassword;

  const onSubmit = async event => {
    try {
      await firebase.updatePassword(newPassword);
      setNewPassword("");
      setConfirmNewPassword("");
      setError(null);
    } catch (error) {
      setError({ ...error });
    }

    event.preventDefault();
  };

  const handleChange = ({ event, setState }) => {
    setState(event.target.value);
    setError(null);
  };
  const onNewPasswordChange = event => {
    handleChange({ event, setState: setNewPassword });
  };
  const onConfirmNewPasswordChange = event => {
    handleChange({ event, setState: setConfirmNewPassword });
  };

  return (
    <>
      <h2>{text.changeYourPassword}</h2>
      <form onSubmit={onSubmit}>
        <NewPasswordInput
          newPassword={newPassword}
          onNewPasswordChange={onNewPasswordChange}
        />
        <ConfirmNewPasswordInput
          confirmNewPassword={confirmNewPassword}
          onConfirmNewPasswordChange={onConfirmNewPasswordChange}
        />
        <SubmitButton isInvalid={isInvalid} />
        {error && <ErrorAlert errorText={error.message} />}
      </form>
    </>
  );
};

export default PasswordChangePage;
export { PasswordChangeForm };
