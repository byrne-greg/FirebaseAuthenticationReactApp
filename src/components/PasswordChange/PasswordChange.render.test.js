import React from "react";
import { render, cleanup } from "@testing-library/react";
import PasswordChange from ".";

afterEach(cleanup);

describe("PasswordForget render tests", () => {
  it("page renders as expected", () => {
    // GIVEN

    // WHEN
    const page = render(<PasswordChange />);

    // THEN
    expect(page.baseElement).toMatchSnapshot();
  });

  xit("form renders with error", () => {});
});
