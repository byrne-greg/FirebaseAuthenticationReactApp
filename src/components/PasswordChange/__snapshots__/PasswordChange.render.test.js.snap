// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordForget render tests page renders as expected 1`] = `
<body>
  <div>
    <div>
      <h1>
        Change Password
      </h1>
      <h2>
        Change your password
      </h2>
      <form>
        <label
          id="new-password-label"
        >
          New Password
        </label>
        <input
          aria-labelledby="new-password-label"
          name="new-password"
          placeholder="New Password"
          type="password"
          value=""
        />
        <label
          id="confirm-new-password-label"
        >
          Confirm New Password
        </label>
        <input
          aria-labelledby="confirm-new-password-label"
          name="confirm-new-password"
          placeholder="Confirm New Password"
          type="password"
          value=""
        />
        <button
          disabled=""
          type="submit"
        >
          Change Password
        </button>
      </form>
    </div>
  </div>
</body>
`;
